virttest/utils_params.py

```diff
@@ -104,3 +105,91 @@ def copy_from_keys(self, keys):
             if self.get(key):
                 new_dict[key] = self.get(key)
         return new_dict
+
+    def get_boolean(self, key, default=False):
+        """
+        Check if a config option is set to a default affirmation or not.
+
+        :param key: config option key
+        :type key: str
+        :param bool default: whether to assume "yes" or "no" if param
+                             does not exist (defaults to False/"no").
+        :return whether option is set to 'yes'
+        :rtype: bool
+        """
+        value = self.get(key, "yes" if default else "no")
+        if value in ("yes", "on", "true"):
+            return True
+        if value in ("no", "off", "false"):
+            return False
+        raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
+
+    def get_numeric(self, key, default=0, target_type=int):
+        """
+        Get numeric value converting to integer if necessary.
+
+        :param str key: config option key
+        :param int default: default numeric value
+        :param type target_type: numeric type to return like int or float
+        :return numerical type `target_type` converted parameter value
+        :rtype: int or float
+        """
+        return target_type(self.get(key, default))
+
+    def get_list(self, key, default="", delimiter=' ', target_type=str):
```

Comment:
`None` would be IMO better delimiter as it is default for `split` (and removes the unwanted empty spaces issue...)

Reply:
Good point, resolved in a future PR.